databricks/koalas/indexes.py

```diff
@@ -174,7 +174,8 @@ def names(self, names: List[Union[str, Tuple[str, ...]]]) -> None:
         if len(internal.index_map) != len(names):
             raise ValueError('Length of new names must be {}, got {}'
                              .format(len(internal.index_map), len(names)))
-        names = [name if isinstance(name, tuple) else (name,) for name in names]
+
+        names = [name if isinstance(name, (tuple, type(None))) else (name,) for name in names]
```

Comment:
It should usually be `is None` but `isinstance(type(None))` is legitimate too. so I think it's fine.